actions/build-docker/action.yaml

```diff
@@ -0,0 +1,34 @@
+name: "Build Docker Image"
+description: "Builds a Docker image"
+inputs:
+  context:
+    description: "Path to the Docker build context."
+    required: true
+  repository:
+    description: "Repository name for the Docker image (defaults to current GitHub repository)."
+    required: false
+    default: ${{ github.repository }}
+  domain:
+    description: "The image's domain."
+    required: true
+  registry:
+    description: Azure Registry to authenticate against.
+    required: true
+runs:
+  using: "composite"
+  steps:
+    - name: Checkout code
+      uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
```

Comment:
Don't we want to add the [`ref` parameter](https://github.com/actions/checkout?tab=readme-ov-file#usage) to the checkout for `PR`'s or other times where we want to use a specific commit `sha`?